lib/segment/analytics/request.rb

```diff
@@ -112,6 +112,11 @@ def send_request(write_key, batch)
 
           [200, '{}']
         else
+          # If `start` is not called, Ruby adds a 'Connection: close' header
+          # too all requests, preventing us from reusing a connection for multiple
+          # HTTP requests
+          @http.start unless @http.started?
```

Comment:
I added this comment because it took me a while to figure this out, and the Ruby docs don't mention it explicitly.

I'll also try raising this on the Ruby issue tracker once I've experimented a bit.

Reply:
Reported upstream: https://bugs.ruby-lang.org/issues/14349